.github/scripts/ipa_version_check.sh

```diff
@@ -1,16 +1,12 @@
-#!/bin/bash  
-set -eou pipefail
+#!/bin/bash      
+set -eou pipefail    
+    
+current_version=$(jq -r '.version' tools/spectral/ipa/package.json)    
   
-current_version=$(jq -r '.version' tools/spectral/ipa/package.json)  
-
-previous_version=$(git show origin/"${BASE_BRANCH}":tools/spectral/ipa/package.json | jq -r '.version')  
+previous_version=$(git show HEAD~1:tools/spectral/ipa/package.json | jq -r '.version' || echo "none")    
```

Comment:
Any reason why we do not use here github defined base branch env vars?
Also you are doing sparse checkouts in actions which might affect history.